Fill.py

```diff
@@ -272,19 +278,26 @@ def distribute_items_restrictive(world: MultiWorld) -> None:
     defaultlocations = locations[LocationProgressType.DEFAULT]
     excludedlocations = locations[LocationProgressType.EXCLUDED]
 
-    prioritylocations_lock = prioritylocations.copy()
+    # can't lock due to accessibility corrections touching things, so we remember which ones got placed and lock later
+    lock_later = []
+
+    def mark_for_locking(location: Location):
+        nonlocal lock_later
+        lock_later.append(location)
 
-    fill_restrictive(world, world.state, prioritylocations, progitempool)
+    # "priority fill"
+    fill_restrictive(world, world.state, prioritylocations, progitempool, swap=False, on_place=mark_for_locking)
     accessibility_corrections(world, world.state, prioritylocations, progitempool)
 
-    for location in prioritylocations_lock:
-        if location.item:
-            location.locked = True
+    for location in lock_later:
+        location.locked = True
```

Comment:
i may be wrong, but to me it looks like accessibility_corrections can remove items from priority locations? so we have to check if location has an item.
```suggestion
        if location.item:
            location.locked = True
```

Reply:
corrections keeps track of all locations that were emptied and then runs another fill_restrictive on them,
so it's save to assume they're filled